lib/iris/cube.py

```diff
@@ -62,7 +62,8 @@ class CubeMetadata(collections.namedtuple('CubeMetadata',
                                            'var_name',
                                            'units',
                                            'attributes',
-                                           'cell_methods'])):
+                                           'cell_methods',
+                                           'cell_measures_and_dims'])):
```

Comment:
This seems quite neat, and it makes ```cube.copy()``` work correctly, but unfortunately, I don't think it is a good idea.

My mental model of ```cube.metadata``` is that it encapsulates only the **"non-dimensional"** information relating to the contained data.
(I do wish that was clearly stated somewhere, but that's another story).

Thus, at present, you can reasonably write code that copies metadata from one cube to another, e.g. like this ...
```
cube_z = cube_a * cube_b
if cube_a.units == '1':
    cube_z.metadata = cube_b.metadata
```
(N.B. I'm not saying that makes **good** sense, but it might in some known case).

Also, I think some existing code actually does this
 \- for example the interpolation scheme  [here](https://github.com/SciTools/iris/blob/b9e6c6d0d9b8b7216197212d989cc7cad568fb2f/lib/iris/analysis/_interpolation.py#L619)


At present, this means that 'metadata' is, basically, everything *except* coords + factories.
For consistency, I think this should *not* include cell_measures.